setup-env/src/actionInput/index.js

```diff
@@ -64,8 +74,71 @@ class ActionInput {
     core.info(`${ENV_VARS.BROWSERSTACK_BUILD_NAME} environment variable set as: ${this.buildName}`);
     core.info(`Use ${ENV_VARS.BROWSERSTACK_BUILD_NAME} environment variable for your build name capability in your tests\n`);
 
+    if (await this.checkIfBStackReRun()) {
+      await this.setBStackRerunEnvVars();
+    }
     core.endGroup();
   }
+
+  async checkIfBStackReRun() {
+    // Using !! ensures that the function returns true or false, regardless of the input values.
+    core.info(`The variables set are: rerunAttempt - ${this.rerunAttempt}, runId - ${this.runId}, repository - ${this.repository}, githubToken - ${this.githubToken}`);
+    if (!this.rerunAttempt || !this.rerunAttempt > 1) {
+      return false;
+    }
+    if (!this.runId || !this.runId > 1 || !this.repository || this.repository === 'none'
+      || !this.githubToken || this.githubToken === 'none' || !this.username || !this.accessKey) {
+      return false;
+    }
+    const triggeringActor = await this.identifyRunFromBStack();
+    core.info(`Triggering actor is - ${triggeringActor}`);
+    return triggeringActor === this.githubApp;
+  }
+
+  async identifyRunFromBStack() {
+    try {
+      const runDetailsUrl = `https://api.github.com/repos/${this.repository}/actions/runs/${this.runId}`;
+      const runDetailsResponse = await axios.get(runDetailsUrl, {
+        headers: {
+          Authorization: `token ${this.githubToken}`,
+          Accept: 'application/vnd.github.v3+json',
+        },
+      });
+
+      return runDetailsResponse.data.triggering_actor?.login;
+    } catch (error) {
+      core.info(`Error getting run details to identify actor of the build: ${error.message}`);
+    }
+  }
+
+  async setBStackRerunEnvVars() {
+    try {
+      // Check if the run was triggered by the BrowserStack rerun bot
+      core.info('The re-run was triggered by the GitHub App from BrowserStack.');
+
+      const browserStackApiUrl = BROWSERSTACK_TEMPLATE.DETAILS_API_URL.replace('{runId}', this.runId);
+
+      // Call BrowserStack API to get the tests to rerun
+      const bsApiResponse = await axios.get(browserStackApiUrl, {
+        auth: {
+          username: this.username.replace("-GitHubAction", ""),
```

Comment:
Get context for this

Reply:
This is for internal instrumentation. More details added https://browserstack.slack.com/archives/CPHUK7ZA5/p1727083890517749?thread_ts=1726650791.006209&cid=CPHUK7ZA5